### ISSUE TITLE
gnome-terminal gets pushed to background

### ISSUE THREAD

**Issue by adityadesaha** [foreshadow/atom-python-run#70]
Hi! I am running atom 1.22.0 on Ubuntu 17.10. If I launch atom-python-run with the terminal specified as gnome-terminal, gnome-terminal launches on the background. I'd like the terminal to pop up in the foreground, like xterm (or terminator).

**Comment by xovertheyearsx** [comment]
Ubuntu 17.10 moved away from Unity and reverted back to Gnome. Gnome now uses version 3 by default and so does 17.10. I happen to use Archlinux with gnome3 and have the same issue, although, it's not neccessarily related to this plug-in. 

It has to do with the shell and how certain windows in the evironment are focused. MS Windows is the only OS that seems to allow the terminal to take focus by default while Mac OS X and Gnome3 seem to push it to background. Not sure if any other systems have this issue as well. 

Let me know if you figure it out. You can always create a fork, pull request, or post a fix here. Although, like I already stated, this issue seems to be related to how gnome3 handles focusing a window.

In short, I don't know how to change/fix it and I'm open to proactive solutions.

**Comment by wickedjargon** [comment]
How do you get Ubuntu to use xterm instead? I tried typing in 'xterm' in the Terminal field in settings but this didn't change anything.

**Comment by xovertheyearsx** [comment]
The terminal field behaves like a list which takes the terminal name and the execution option. If you're using unity, you should be okay.

make sure xterm is installed (it usually is by default in ubuntu)
- press the `meta key` (usually the windows key)
- in the search field, type in `xterm`

If it's installed, you'll see it in the results. If not, you'll have to install it.

In Atom, go to `settings` -> `atom-python-run` -> `settings` -> `terminal`
![i.e. xterm, -e](https://i.imgur.com/IXNQKlP.png)

That's all there is to it.

**Comment by xovertheyearsx** [comment]
@FredhRodrigues -- [README](https://github.com/foreshadow/atom-python-run/wiki/00-Readme) -- #69 

### **Terminal opening in background (Ubuntu 16) #73**  

> Hello guys. I was using everything without a problem, then I don't know if I hit the wrong key or changed the wrong lines in config.cson but now this started to happen. At first I thought f5 was not working anymore then I saw a dozen terminal windows in the back.
> 
> This is really annoying. Is there anyway to make it open in the foreground again? Thanks in advance.

Even though issue #73 takes place in Ubuntu 16.xx, which I find interesting because it uses Unity as its WM and DE, it can be cataloged here since it resembles this issue, #70. Issue #73 will be merged with issue #70 to help isolate this general issue.

**Comment by FredhRodrigues** [comment]
I was not sure where to post it since it was in a different version of Ubuntu so I figured it would be best to start a new thread. I did check issue 70 before posting and I can use xterm but that doesn't explain why this started to happen to terminal so suddenly.

**Comment by xovertheyearsx** [comment]
@FredhRodrigues Both Unity and Gnome use gnome-terminal as their default terminal. So it's okay. Hopefully, someone can help figure it out. I still don't have a solution. If one is resolved, the other won't be, and that will allow some focus to remain on that issue. It might even give a hint or a clue that could help resolve it.

**Comment by FredhRodrigues** [comment]
Reinstalling Atom (purge and install) seemed to have solved the problem, everything is working fine now. Thanks for the help.
